app/src/consumer/parser_consumer.c

```diff
@@ -360,6 +366,69 @@ __Z_INLINE parser_error_t parser_getItemTx(const parser_context_t *ctx,
                 }
             }
             break;
+        case stakingWithdraw:
+            switch (displayIdx) {
+                case 0: {
+                    snprintf(outKey, outKeyLen, "Type");
+                    *pageCount = 1;
+                    return parser_getType(ctx, outVal, outValLen);
+                }
+                case 1: {
+                    snprintf(outKey, outKeyLen, "Address");
+                    return parser_printAddress(&parser_tx_obj.oasis.tx.body.stakingWithdraw.from,
+                                               outVal, outValLen, pageIdx, pageCount);
+                }
+                case 2: {
+                    snprintf(outKey, outKeyLen, "Amount");
+                    return parser_printQuantity(&parser_tx_obj.oasis.tx.body.stakingWithdraw.amount,
+                                                outVal, outValLen, pageIdx, pageCount);
+                }
+                case 3: {
+                    // ??? displayIdx == 1 && parser_tx_obj.oasis.tx.has_fee
+                    snprintf(outKey, outKeyLen, "Fee");
+                    return parser_printQuantity(&parser_tx_obj.oasis.tx.fee_amount, outVal, outValLen, pageIdx,
+                                                pageCount);
+                }
+                case 4: {
+                    snprintf(outKey, outKeyLen, "Gas limit");
+                    uint64_to_str(outVal, outValLen, parser_tx_obj.oasis.tx.fee_gas);
+                    *pageCount = 1;
+                    return parser_ok;
+                }
+            }
+            break;
+        case stakingAllow:
+            switch (displayIdx) {
+                case 0: {
+                    snprintf(outKey, outKeyLen, "Type");
+                    *pageCount = 1;
+                    return parser_getType(ctx, outVal, outValLen);
+                }
+                case 1: {
+                    snprintf(outKey, outKeyLen, "Beneficiary");
+                    return parser_printAddress(&parser_tx_obj.oasis.tx.body.stakingAllow.beneficiary,
+                                               outVal, outValLen, pageIdx, pageCount);
+                }
+                case 2: {
+                    // FIXME: Handle stakingAllow.is_negative in amount_change
```

Comment:
Yes, please, change the label to "Amount change" and display the sign (i.e. `+` / `-` ) in from of the amount, depending on the value of the `stakingAllow.is_negative` boolean.

Reply:
this is not the correct place to do this.. 
Sign handling should go in `parser_printQuantity`

Reply:
the reason is.. we need to handle paging correctly and we don't want to sign to be misplaced when doing this

Reply:
Where do we have to handle the sign?

Reply:
I just updated the test cases and now some of them are failing. I have to handle the sign to make them pass.

Reply:
I fixed the tests, and now all is green. I know the way I handle the sign is not the correct. Could you tell me where to handle it?